macOS/DuckDuckGo/History/View/HistoryViewDataProvider.swift

```diff
@@ -0,0 +1,271 @@
+//
+//  HistoryViewDataProvider.swift
+//
+//  Copyright © 2025 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import BrowserServicesKit
+import Foundation
+import History
+import HistoryView
+
+protocol HistoryBurning: AnyObject {
+    func burn(_ visits: [Visit], animated: Bool) async
+}
+
+final class FireHistoryBurner: HistoryBurning {
+    let fireproofDomains: DomainFireproofStatusProviding
+    let fire: () async -> Fire
+
+    init(fireproofDomains: DomainFireproofStatusProviding = FireproofDomains.shared, fire: (() async -> Fire)? = nil) {
+        self.fireproofDomains = fireproofDomains
+        self.fire = fire ?? { @MainActor in FireCoordinator.fireViewModel.fire }
+    }
+
+    func burn(_ visits: [Visit], animated: Bool) async {
+        await withCheckedContinuation { continuation in
+            Task { @MainActor in
+                await fire().burnVisits(visits, except: fireproofDomains, isToday: animated) {
```

Comment:
Could there be a problem with the `burnVisits` call here? From the stacktrace, the `animates` flag is true when the range is today or all, if it is all, could it lead to a problem?

Reply:
This isn't a bug actually :) The behavior follows history clearing behavior from the main menu – clearing data for all history needs to close all tabs. The same for today (because we're clearing data for the tabs opened today, i.e. tabs open right now, therefore they must be closed). Data cleared for any previous day doesn't close tabs.

We will update the copy of the burn dialog to reflect this (see [Copy Review discussion](https://app.asana.com/0/38424471409662/1209167289674669) for reference) - the implementation is [already in place](https://github.com/duckduckgo/apple-browsers/commit/31704cb5d238166b45c00629d920468aec18632e), but due to changes to history delete dialog, the commit is added on another branch.